### PR TITLE
Delete unused code & add tests

### LINKED ISSUE #12: Simplify hyper-nested tests structure
Currently there is a "tests" subfolder within the "tests" folder, which results in some absurd paths like: `tests/tests/test_unit/test_general`.
This structure could use some refactoring.

### PR DESCRIPTION
**Pull request by adamltyson**
## Description

**What is this PR**

- [ ] Bug fix
- [ ] Addition of a new feature
- [x] Other

**Why is this PR needed?**
There is a lot of code in this repo that is no longer used by any BrainGlobe tool. Keeping it in the repo increases the maintenance burden. 

**What does this PR do?**
Removes unused code and adds some more tests for what remains.

Also closes #12 

## How has this PR been tested?

Please explain how any new code has been tested, and how you have ensured that no existing functionality has changed.

## Is this a breaking change?

Hopefully not. I've checked other BG repos for usage of deleted code, and nothing came up. This process isn't foolproof though, so we may just have to release and hope for the best! This is a fairly dangerous approach, so happy to hear of alternatives.

## Does this PR require an update to the documentation?

If any features have changed, or have been added. Please explain how the documentation has been updated (and link to the associated PR). See [here](https://docs.cellfinder.info/for-developers/documentation) for details.

## Checklist:

- [x] The code has been tested locally
- [x] Tests have been added to cover all new functionality (unit & integration)
- [x] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

**Review by willGraham01 — COMMENTED** [review]
I can't speak for all of the tests as I'm not entirely sure what they're all meant to be doing (maybe we need an issue for adding docstrings to this package, maybe a docs build that can be handed off to the website?).

But otherwise, more tests = better. One remark on a comment that's been removed but doesn't seem to have been addressed?

File: brainglobe_utils/IO/cells.py [hunk 1]
```diff
@@ -29,11 +29,9 @@ def get_cells(
     cells_only: bool = False,
     cell_type: Optional[int] = None,
 ):
-    # TODO: implement csv read
```
[the remark this answers is not in the document]
Comment (adamltyson) [reply brainglobe_utils/IO/cells.py:32]: No, but there's no need to.

**Review by IgorTatarnikov — APPROVED** [review]
@willGraham01 beat me to the punch, but everything looks good to me!

**Comment by adamltyson** [comment]
@willGraham01 I've raised  https://github.com/brainglobe/brainglobe-utils/issues/19 Adding docstrings is something the entirety of BG sorely needs.

**Comment by adamltyson** [comment]
Thanks both. Going to release a new version, so yell at me if anything breaks!

**Pull request merged**